src/Nix/Value.hs

```diff
@@ -266,6 +267,7 @@ hoistNValueF lft =
 
 -- * @__NValue'__@: forming the (F(A))
 
+type NvConstraint f = (Comonad f, Applicative f) 
```

Comment:
This constraint would be used in a lot of places. So people would arrive to it & wonder why it is there. It would be useful to provide a minimal explanation to why it was needed, it is a nice place to explain both the bidirectional pattern solution and what this constraint does.

Something of a kind: "to allow/enforce bidirectionality in patterns".

Reply:
Thank you @Anton-Latukha  for pointing these problems out.

`NV` as short hand for Nix Value to have uppercase would make sense too.
`NVConstraint` does seem to be a proper name in the current situation.

And documentation definitely needed in such places. 
And I need to sort out the documentation for code changes.

I would adjust the these two problems now.

Reply:
renamed to `NVConstraint` and adjust the doc in new commits done